wordpress-mobile/WordPress-Android
# [Dependency Updates] Main Batch - AndroidX Core

Parent #17551
Closes #17561

This PR updates:
- [x] `androidxArchCoreVersion` to [2.1.0](https://developer.android.com/jetpack/androidx/releases/arch-core#2.1.0) (via #17642).
- [x] `androidxLifecycleVersion` to [2.5.1](https://developer.android.com/jetpack/androidx/releases/lifecycle#2.5.1) (via #17643).
- [x] `androidxCoreVersion` to [1.8.0](https://developer.android.com/jetpack/androidx/releases/core#1.8.0) (via #17691).
- [x] `androidxFragmentVersion` to [1.5.5](https://developer.android.com/jetpack/androidx/releases/fragment#1.5.5) (via #17696).
- [x] `androidxAppcompatVersion` to [1.4.2](https://developer.android.com/jetpack/androidx/releases/appcompat#1.4.2) (via  #17743).

Note that this is the parent branch were all the above PRs were targeting. As such:
1. The code review part was already done as part of the individual PRs above. Thus, no Dev is to review this PR.
2. This PR's sole purpose is making sure these changes are thoroughly tested by a QE.

-----

PS: @pachlava I added you as the main reviewer, that is, in addition to the @wordpress-mobile/apps-infrastructure team itself, since I just wanted someone from the QE team to thoroughly test and sign-off on that change for WPAndroid.

-----

To test:

- Thoroughly smoke test both, the WordPress and Jetpack apps, and see if they both work as expected.
- In addition to the above smoke test, per the linked PRs above, you could follow the individual test instructions included in them and focus on the specifics of what to test per update.

-----

### Merge instructions

- [x] Wait for #17743 PR to be approved and merged to this `deps/main-batch-androidx-core` branch.
- [x] Prepare and update PR to a `Ready for review` state.
- [x] Assign a QE reviewer.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage of screens and core functionality as these `AndroidX Core` libraries are being added as a transitive dependency across lots of `androidx` and `com.google` and `other` libraries.
    - Some of the transitive dependencies added might be causing some kind of misbehaviour.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - `N/A`

-----

PR submission checklist:
